### PR TITLE
Add new environment variables for diagnostic service

### PR DESCRIPTION
**Pull request by zhang-hua**
# Description and Motivation 

...

# Change type 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [x] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

---
This change is [Reviewable](https://reviewable.io/reviews/azure/pcs-cli/332)

### CODE REVIEW

**Comment by zhang-hua** [comment]
Will rebase code with master branch after PR #331 was merged.

**Comment by molly-moen** [comment]
I forgot we also need to add PCS_SOLUTION_NAME as a variable (I thought it was added but it is still in PR from the interns: https://github.com/Azure/pcs-cli/pull/321/files), and send it to the diagnostics service

File: solutions/devicesimulation/single-vm/docker-compose.yml [hunk 1]
```diff
@@ -101,8 +101,11 @@ services:
       - PCS_AUTH_AUDIENCE
       - PCS_AUTH_REQUIRED
       - PCS_CORS_WHITELIST
+      - PCS_CLOUD_TYPE
+      - PCS_SUBSCRIPTION_ID
       - PCS_DEPLOYMENT_ID
       - PCS_SOLUTION_TYPE
+      - PCS_IOTHUB_NAME
       - PCS_DIAGNOSTICS_ENDPOINT_URL
```
Comment (molly-moen) [inline solutions/devicesimulation/single-vm/docker-compose.yml]: can you also add PCS_DIAGNOSTICS_ENDPOINT_URL as a var for telemetry microservice?

**Comment by ppathan** [comment]
@zhang-hua Can you resolve the conflicts?

File: solutions/devicesimulation/single-vm/docker-compose.yml (hunk 1 above)
Comment (zhang-hua) [reply solutions/devicesimulation/single-vm/docker-compose.yml]: yes, I will add it.

Comment (zhang-hua) [reply solutions/devicesimulation/single-vm/docker-compose.yml]: @molly-moen , is it only required for telemetry service? what about the other services?

File: solutions/remotemonitoring/single-vm/docker-compose.dotnet.yml [hunk 2]
```diff
@@ -85,7 +85,6 @@ services:
       - default_net
     depends_on:
       - auth
-      - diagnostics
```
Comment (molly-moen) [inline solutions/remotemonitoring/single-vm/docker-compose.dotnet.yml]: why is this not needed?

Comment (zhang-hua) [reply solutions/remotemonitoring/single-vm/docker-compose.dotnet.yml]: It introduced loop dependency:  telemetry->diagnostics->config->telemetry.

File: src/deploymentmanager.ts [hunk 3]
```diff
@@ -553,6 +559,17 @@ export class DeploymentManager implements IDeploymentManager {
 
         console.log('Copy the following environment variables to /scripts/local/.env file: \n\ %s', `${chalk.cyan(data.join('\n'))}`);
     }
+
+    // Internal cloud names for diagnostics
+    private getCloudType(environmentName: string): string {
+        const cloudTypeMaps = {
```
Comment (ppathan) [inline src/deploymentmanager.ts]: Do you want to move the const out? Also why can't we send enum.toString() value. Why do we need to do a conversion?

Comment (zhang-hua) [reply src/deploymentmanager.ts]: The list is provided by internal team for data analysis.

File: solutions/devicesimulation/single-vm/docker-compose.yml [hunk 4]
```diff
@@ -102,8 +102,11 @@ services:
       - PCS_AUTH_AUDIENCE
       - PCS_AUTH_REQUIRED
       - PCS_CORS_WHITELIST
+      - PCS_CLOUD_TYPE
+      - PCS_SUBSCRIPTION_ID
       - PCS_DEPLOYMENT_ID
       - PCS_SOLUTION_TYPE
+      - PCS_IOTHUB_NAME
```
Comment (molly-moen) [inline solutions/devicesimulation/single-vm/docker-compose.yml]: also add PCS_SOLUTION_NAME

File: src/k8smanager.ts [hunk 5]
```diff
@@ -193,6 +193,9 @@ export class K8sManager implements IK8sManager {
         configMap.data['security.application.secret'] = genPassword();
         configMap.data['azure.maps.key'] = this._config.AzureMapsKey ? this._config.AzureMapsKey : '';
         configMap.data['iothub.connstring'] = this._config.IoTHubConnectionString;
+        configMap.data['diagnostics.cloud.type'] = this._config.CloudType;
+        configMap.data['diagnostics.subscription.id'] = this._config.SubscriptionId;
+        configMap.data['diagnostics.iothub.name'] = this._config.IotHubName;
```
Comment (molly-moen) [inline src/k8smanager.ts]: does solution name need to be added here?

File: src/deploymentmanager.ts [hunk 6]
```diff
@@ -230,6 +230,9 @@ export class DeploymentManager implements IDeploymentManager {
                     config.AzureStorageEndpointSuffix = storageEndpointSuffix;
                     // If we are under the plan limit then we should have received a query key
                     config.AzureMapsKey = outputs.azureMapsKey.value;
+                    config.CloudType = this.getCloudType(this._environment.name);
+                    config.IotHubName = outputs.iotHubHostName;
+                    config.SubscriptionId = outputs.subscriptionId;
```
Comment (molly-moen) [inline src/deploymentmanager.ts]: does solution name need to be added here?

**Pull request merged**